pyinterprod/pronto/protein.py

```diff
@@ -289,3 +290,90 @@ def import_proteins(ora_url: str, pg_url: str):
 
     pg_con.close()
     logger.info("complete")
+
+
+def import_protein_pubmed(ip_url: str, swp_url: str, pg_url: str):
+    logger.info("populating protein2publication")
+    pg_con = psycopg.connect(**url2dict(pg_url))
+    with pg_con.cursor() as pg_cur:
+        pg_cur.execute("DROP TABLE IF EXISTS protein2publication")
+        pg_cur.execute(
+            """
+            CREATE TABLE protein2publication (
+                ACCESSION VARCHAR(15) NOT NULL,
+                PUBMED_ID VARCHAR(15) NOT NULL
+            )
+            """
+        )
+
+        ora_con = oracledb.connect(ip_url)
+        with ora_con.cursor() as ora_cur:
+            ora_cur.execute(
+                """
+                SELECT PROTEIN_AC
+                FROM INTERPRO.PROTEIN
+                WHERE DBCODE = 'S'
+                """
+            )
+            swp_acc = [acc for acc, in ora_cur.fetchall()]
+
+        sql = """
+            INSERT INTO protein2publication
+                (accession, pubmed_id)
+            VALUES (%s, %s)
+        """
+        reg_pubmed = re.compile(r"(PubMed:(\d+))")
+
+        ora_con = oracledb.connect(swp_url)
+        ora_cur = ora_con.cursor()
+        swp2pmid = set()
+
+        for i in range(0, len(swp_acc), 1000):
+            swp_acc_chunk = swp_acc[i:i+1000]
+            args = [':' + str(c+1) for c in range(len(swp_acc_chunk))]
+            ora_cur.execute(
+                f"""
+                    SELECT E.ACCESSION, NVL(B.TEXT, SS.TEXT) AS TEXT
+                    FROM SPTR.DBENTRY E
+                    INNER JOIN SPTR.COMMENT_BLOCK B ON E.DBENTRY_ID = B.DBENTRY_ID
+                        AND B.COMMENT_TOPICS_ID = 2
+                    INNER JOIN SPTR.CV_COMMENT_TOPICS CT 
+                        ON B.COMMENT_TOPICS_ID = CT.COMMENT_TOPICS_ID
+                    LEFT OUTER JOIN SPTR.COMMENT_STRUCTURE S
+                        ON B.COMMENT_BLOCK_ID = S.COMMENT_BLOCK_ID
+                        AND S.CC_STRUCTURE_TYPE_ID = 1
+                    LEFT OUTER JOIN SPTR.COMMENT_SUBSTRUCTURE SS
+                        ON S.COMMENT_STRUCTURE_ID = SS.COMMENT_STRUCTURE_ID
+                    WHERE E.ACCESSION IN ({','.join(args)})
+                      AND E.ENTRY_TYPE IN (0, 1)     
```

Comment:
My bad, the query I posted on Jira retrieves records for all UniProtKB entries (Swiss-Prot + TrEMBL). This change restricts the query to Swiss-Prot entries:

```suggestion
                      AND E.ENTRY_TYPE IN = 0
```